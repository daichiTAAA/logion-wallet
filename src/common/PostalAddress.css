.PostalAddress {
    padding-left: 5px;
    padding-right: 5px;
}

.PostalAddress .PostalAddressCol {
    padding-left: 10px;
    padding-right: 10px;
}

.PostalAddress h3 {
    margin-top: 15px;
    margin-bottom: 15px;
    text-align: left !important;
    margin-left: -5px;
}
