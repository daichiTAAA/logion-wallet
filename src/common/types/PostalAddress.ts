export default interface PostalAddress {
    company?: string,
    line1: string,
    line2: string,
    postalCode: string,
    city: string,
    country: string
}
