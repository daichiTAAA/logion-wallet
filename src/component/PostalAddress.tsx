import React from 'react';
import Form from 'react-bootstrap/Form';
import FormGroup from 'react-bootstrap/FormGroup';

import PostalAddressType from './types/PostalAddress';

export interface Props {
    address: PostalAddressType
}

export default function PostalAddress(props: Props) {
    return (
        <>
            <FormGroup>
                <Form.Label>Line 1</Form.Label>
                <Form.Control type="text" value={ props.address.line1 } readOnly />
            </FormGroup>
            <FormGroup>
                <Form.Label>Line 2</Form.Label>
                <Form.Control type="text" value={ props.address.line2 } readOnly />
            </FormGroup>
            <FormGroup>
                <Form.Label>Postal code</Form.Label>
                <Form.Control type="text" value={ props.address.postalCode } readOnly />
            </FormGroup>
            <FormGroup>
                <Form.Label>City</Form.Label>
                <Form.Control type="text" value={ props.address.city } readOnly />
            </FormGroup>
            <FormGroup>
                <Form.Label>Country</Form.Label>
                <Form.Control type="text" value={ props.address.country } readOnly />
            </FormGroup>
        </>
    )
}
