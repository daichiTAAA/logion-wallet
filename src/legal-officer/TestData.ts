import moment from 'moment';
import { toIsoString } from '../logion-chain/datetime';
import { ProtectionRequest, DEFAULT_LEGAL_OFFICER } from './Types';

export const PENDING_PROTECTION_REQUESTS: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        decisions: [
            {
                legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
                status: 'PENDING',
                rejectReason: null,
                decisionOn: null
            }
        ],
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z'))
    }
];

export const PROTECTION_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        decisions: [
            {
                legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
                status: 'ACCEPTED',
                rejectReason: null,
                decisionOn: toIsoString(moment('2021-06-10T11:40:00.000Z'))
            }
        ],
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z'))
    }
];
