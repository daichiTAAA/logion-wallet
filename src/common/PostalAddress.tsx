import { Row, Col } from "react-bootstrap";
import { PostalAddress as PostalAddressType } from "@logion/client";

import { BackgroundAndForegroundColors } from "./ColorTheme";
import ComparableField from "./ComparableField";
import "./PostalAddress.css";

export interface Props {
    postalAddress?: PostalAddressType
    otherPostalAddress?: PostalAddressType
    colors: BackgroundAndForegroundColors
    squeeze: boolean
    noComparison: boolean
}

export default function PostalAddress(props: Props) {

    interface FieldProps {
        id: string
        label: string
        field: (postalAddress: PostalAddressType) => string
        noComparison: boolean
    }

    function ComparablePostalAddressField(fieldProps: FieldProps) {

        return (
            <ComparableField
                id={ fieldProps.id }
                label={ fieldProps.label }
                data={ props.postalAddress }
                otherData={ props.otherPostalAddress }
                field={ fieldProps.field }
                colors={ props.colors }
                squeeze={ props.squeeze }
                noComparison={ props.noComparison }
            />
        );
    }

    return (
        <div className="PostalAddress">
            <h3>Address</h3>
            <Row>
                <Col className="PostalAddressCol" md={ 12 }>
                    <ComparablePostalAddressField
                        id="line1"
                        label="Line1"
                        field={ address => address.line1 }
                        noComparison={ props.noComparison }
                    />
                </Col>
            </Row>
            <Row>
                <Col className="PostalAddressCol" md={ 12 }>
                    <ComparablePostalAddressField
                        id="line2"
                        label="Line2"
                        field={ address => address.line2 }
                        noComparison={ props.noComparison }
                    />
                </Col>
            </Row>
            <Row>
                <Col className="PostalAddressCol" md={ 4 }>
                    <ComparablePostalAddressField
                        id="postalCode"
                        label="Postal Code"
                        field={ address => address.postalCode }
                        noComparison={ props.noComparison }
                    />
                </Col>
                <Col className="PostalAddressCol" md={ 8 }>
                    <ComparablePostalAddressField
                        id="city"
                        label="City"
                        field={ address => address.city }
                        noComparison={ props.noComparison }
                    />
                </Col>
            </Row>
            <Row>
                <Col className="PostalAddressCol" md={ 12 }>
                    <ComparablePostalAddressField
                        id="country"
                        label="Country"
                        field={ address => address.country }
                        noComparison={ props.noComparison }
                    />
                </Col>
            </Row>
        </div>
    )
}
