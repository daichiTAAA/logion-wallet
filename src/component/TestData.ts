import Identity from './types/Identity';
import PostalAddress from './types/PostalAddress';
import { ColorTheme, rgbaToHex } from '../component/ColorTheme';

export const DEFAULT_IDENTITY: Identity = {
    firstName: "John",
    lastName: "Doe",
    email: "john.doe@logion.network",
    phoneNumber: "+1234",
};

export const DEFAULT_ADDRESS: PostalAddress = {
    line1: "Place de le République Française, 10",
    line2: "boite 15",
    postalCode: "4000",
    city: "Liège",
    country: "Belgium",
};

export const COLOR_THEME: ColorTheme = {
    type: 'light',
    shadowColor: rgbaToHex('#3b6cf4', 0.1),
    dashboard: {
        background: '#eff3fe',
        foreground: '#000000',
    },
    sidebar: {
        background: '#ffffff',
        foreground: '#000000',
        activeItemBackground: '#d8e2fd',
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#000000', 0.6),
        foreground: '#000000',
        background: 'eff3fe',
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#ffffff',
        foreground: '#000000',
        link: rgbaToHex('#3b6cf4', 0.15),
        altBackground: '#eff3fe',
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    buttons: {
        secondary: {
            background: '#eff3fe',
            foreground: '#000000',
        }
    },
    select: {
        background: '#ffffff',
        foreground: '#000000',
        menuBackgroundColor: '#ffffff',
        selectedOptionBackgroundColor: '#ffffff',
    },
    table: {
        background: '#0c163d',
        foreground: '#ffffff',
        header: {
            background: '#0c163d',
            foreground: '#ffffff',
        },
        row: {
            background: '#152665',
            foreground: '#ffffff',
        }
    },
    tabs: {
        background: '#0c163d',
        foreground: '#ffffff',
        inactiveTab: {
            background: '#d8e2fd',
            foreground: '#ffffff',
        },
        borderColor: '#3b6cf4',
    },
    dialog: {
        background: '#eff3fe',
        foreground: '#000000',
        borderColor: '#e6007a',
    },
};
