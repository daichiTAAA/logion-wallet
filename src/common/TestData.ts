import { Coin, CoinBalance, SYMBOL } from '../logion-chain/Balances';
import { PrefixedNumber, ATTO } from '../logion-chain/numbers';

import Identity from './types/Identity';
import PostalAddress from './types/PostalAddress';
import { Transaction } from './types/ModelTypes';
import { ColorTheme, rgbaToHex } from './ColorTheme';

export const DEFAULT_LEGAL_OFFICER = "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"; // Alice
export const ANOTHER_LEGAL_OFFICER = "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"; // Bob
export const A_THIRD_LEGAL_OFFICER = "5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y"; // Charlie

export const DEFAULT_IDENTITY: Identity = {
    firstName: "John",
    lastName: "Doe",
    email: "john.doe@logion.network",
    phoneNumber: "+1234",
};

export const DEFAULT_ADDRESS: PostalAddress = {
    line1: "Place de le République Française, 10",
    line2: "boite 15",
    postalCode: "4000",
    city: "Liège",
    country: "Belgium",
};

export const COLOR_THEME: ColorTheme = {
    type: 'light',
    shadowColor: rgbaToHex('#3b6cf4', 0.1),
    dashboard: {
        background: '#eff3fe',
        foreground: '#000000',
    },
    sidebar: {
        background: '#ffffff',
        foreground: '#000000',
        activeItemBackground: '#d8e2fd',
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#000000', 0.6),
        foreground: '#000000',
        background: 'eff3fe',
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#ffffff',
        foreground: '#000000',
        link: rgbaToHex('#3b6cf4', 0.15),
        altBackground: '#eff3fe',
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    recoveryItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    buttons: {
        secondary: {
            background: '#eff3fe',
            foreground: '#000000',
        }
    },
    select: {
        background: '#ffffff',
        foreground: '#000000',
        menuBackgroundColor: '#ffffff',
        selectedOptionBackgroundColor: '#ffffff',
    },
    table: {
        background: '#0c163d',
        foreground: '#ffffff',
        header: {
            background: '#0c163d',
            foreground: '#ffffff',
        },
        row: {
            background: '#152665',
            foreground: '#ffffff',
        }
    },
    tabs: {
        background: '#0c163d',
        foreground: '#ffffff',
        inactiveTab: {
            background: '#d8e2fd',
            foreground: '#ffffff',
        },
        borderColor: '#3b6cf4',
    },
    dialog: {
        background: '#eff3fe',
        foreground: '#000000',
        borderColor: '#e6007a',
    },
};

export const DEFAULT_COIN: Coin = {
    id: 'log',
    iconId: 'log',
    iconType: 'png',
    name: "Logion",
    symbol: SYMBOL
};

export const DEFAULT_COIN_BALANCE: CoinBalance = {
    coin: DEFAULT_COIN,
    balance: new PrefixedNumber("42", ATTO),
    level: 0.1,
};

export const DEFAULT_TRANSACTION: Transaction = {
    from: "from",
    to: "to",
    pallet: "pallet",
    method: "method",
    transferValue: "420",
    tip: "0",
    fee: "125000",
    reserved: "0",
    total: "125420",
    createdOn: "2021-07-28T12:30:00.000",
    type: "Sent",
}
