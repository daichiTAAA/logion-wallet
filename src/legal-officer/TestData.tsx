import moment from 'moment';
import { toIsoString } from '@logion/node-api/dist/datetime';
import { ProtectionRequest } from '@logion/client/dist/RecoveryClient';

import { DEFAULT_LEGAL_OFFICER, ANOTHER_LEGAL_OFFICER } from "../common/TestData";

export const PENDING_PROTECTION_REQUESTS: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER,
        decision: {
            rejectReason: null,
            decisionOn: null,
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
        isRecovery: false,
        addressToRecover: null,
        status: "PENDING"
    }
];

export const PROTECTION_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER,
        decision: {
                rejectReason: null,
                decisionOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
                locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
        isRecovery: false,
        addressToRecover: null,
        status: "ACTIVATED"
    }
];

export const RECOVERY_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER,
        decision: {
            rejectReason: null,
            decisionOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
            locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
        isRecovery: true,
        addressToRecover: "an-address",
        status: "PENDING"
    }
];
