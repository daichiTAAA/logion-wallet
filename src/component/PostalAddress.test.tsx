import PostalAddress from './PostalAddress';
import { render } from '../tests';

import { DEFAULT_ADDRESS } from './TestData';

test("renders", () => {
    const tree = render(<PostalAddress address={ DEFAULT_ADDRESS } />);
    expect(tree).toMatchSnapshot();
});
