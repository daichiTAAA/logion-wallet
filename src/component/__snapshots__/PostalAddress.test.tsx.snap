// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Line 1
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="Place de le République Française, 10"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Line 2
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="boite 15"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Postal code
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="4000"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      City
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="Liège"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Country
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="Belgium"
    />
  </div>,
]
`;
